applications/debug_tools/i2c_scanner.c

```diff
@@ -0,0 +1,101 @@
+#include <furi.h>
+#include <furi-hal.h>
+#include <gui/view.h>
+#include <gui/view_dispatcher.h>
+#include <gui/modules/submenu.h>
+#include <gui/gui.h>
+
+static const char header[] = "I2C C0=SCL C1=SDA";
+
+typedef struct {
+    Gui* gui;
+    ViewDispatcher* view_dispatcher;
+    Submenu* submenu;
+    bool device_found[128];
```

Comment:
We actually have m-lib for C containers: it got arrays and `string_t` that suites perfectly for this task.